### DISCUSSION TITLE
Restrict elaboration to a region of the image

### DISCUSSION THREAD

**Discussion by Fr4nZ82** [ImageMagick/ImageMagick#2841] [Help] [answered]
I need to process a large amount of scans of dot matrix printed documents in order to optimize them for reading with an ocr engine. I used imagemagick to make sure that there are no white spaces between the points of the matrix, so the ocr engine works much better. The problem is performance, pdfs are scanned at 600dpi and processing takes too long. I would like to limit the processing only to the area affected by the zonal ocr, I tried with the "-region" operator but even if it works, the processing takes the same time. This is the command used by the windows command line:
`convert -density "601.6x600" -units pixelsperinch -monochrome files\1.pdf -region 2000x200+2500+2100 -negate -morphology Close Disk -negate -compress zip r.pdf`

https://stackoverflow.com/questions/64892043/imagemagick-restrict-elaboration-to-a-region-of-the-image

**Comment by snibgo** [comment]
That isn't a valid command. It should be "-morphology", with a hyphen. I assume a simple copy/paste error.

Perhaps you can simplify "-negate -morphology Close Disk -negate", eg to "-morphology Close Disk".

Do you need to write the entire image(s) to an output PDF? If not, then use "-crop" instead of "-region".

**Reply by Fr4nZ82** [reply]
yes, i'm working from nodejs child_process.spawn, the command params is splitted in an array and i make a mistake copying it... I need entire pdf as result

**Reply by Fr4nZ82** [reply]
I had thought of using crop, but i didn't understand how to merge cropped over original after elaboration

**Reply by Fr4nZ82** [reply]
tried using -crop with same coordinates of -region and the result was a blank page. Tried modifyng the cropped area to 200x200+20+20 and in the result pdf the cropped area is right, but is shifted at the end of the page so i guess that with right coordinates the result is shifted out of the page.
Please help me...
The complete command tested from the windows cli is:

 `convert -density 601.6x600 -units pixelsperinch -monochrome 1.pdf -region 2500x400+2350+1990 -negate -morphology Thinning "17x17+8+8: -,-,-,-,-,0,0,0,0,0,0,0,-,-,-,-,- -,-,-,-,0,0,0,0,0,0,0,0,0,-,-,-,- -,-,-,0,0,0,0,0,0,0,0,0,0,0,-,-,- -,-,0,0,0,0,-,-,-,-,-,0,0,0,0,-,- -,0,0,0,0,-,-,-,-,-,-,-,0,0,0,0,- 0,0,0,0,-,-,-,-,-,-,-,-,-,0,0,0,0 0,0,0,-,-,-,-,-,-,-,-,-,-,-,0,0,0 0,0,0,-,-,-,-,-,-,-,-,-,-,-,0,0,0 0,0,0,-,-,-,-,-,1,-,-,-,-,-,0,0,0 0,0,0,-,-,-,-,-,-,-,-,-,-,-,0,0,0 0,0,0,-,-,-,-,-,-,-,-,-,-,-,0,0,0 0,0,0,0,-,-,-,-,-,-,-,-,-,0,0,0,0 -,0,0,0,0,-,-,-,-,-,-,-,0,0,0,0,- -,-,0,0,0,0,-,-,-,-,-,0,0,0,0,-,- -,-,-,0,0,0,0,0,0,0,0,0,0,0,-,-,- -,-,-,-,0,0,0,0,0,0,0,0,0,-,-,-,- -,-,-,-,-,0,0,0,0,0,0,0,-,-,-,-,-" -morphology Thinning "13x13+6+6: -,-,0,0,0,0,0,0,0,0,0,-,- -,0,0,0,0,0,0,0,0,0,0,0,- 0,0,0,-,-,-,-,-,-,-,0,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,1,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,0,-,-,-,-,-,-,-,0,0,0 -,0,0,0,0,0,0,0,0,0,0,0,- -,-,0,0,0,0,0,0,0,0,0,-,-" -morphology Dilate Disk -morphology Thinning "9x9+4+4: -,-,0,0,0,0,0,-,- -,0,0,0,0,0,0,0,- 0,0,0,-,-,-,0,0,0 0,0,-,-,-,-,-,0,0 0,0,-,-,1,-,-,0,0 0,0,-,-,-,-,-,0,0 0,0,0,-,-,-,0,0,0 -,0,0,0,0,0,0,0,- -,-,0,0,0,0,0,-,-" -morphology Erode Disk -morphology Thinning "9x9+4+4: -,-,0,0,0,0,0,-,- -,0,0,0,0,0,0,0,- 0,0,0,-,-,-,0,0,0 0,0,-,-,-,-,-,0,0 0,0,-,-,1,-,-,0,0 0,0,-,-,-,-,-,0,0 0,0,0,-,-,-,0,0,0 -,0,0,0,0,0,0,0,- -,-,0,0,0,0,0,-,-" -negate -compress zip result.pdf`

IM version: ImageMagick 7.0.10-39 Q8 x64 2020-11-17

i also share a copy of original and result pdf.
[files](https://drive.google.com/drive/folders/1RaxUGd9kzizTaBU82VsxdGZttnoe-rIh?usp=sharing)

**Comment by snibgo** [comment, marked as answer]
These two commands (Windows syntax) have almost exactly the same result:
```
magick in.tiff -region 100x100+500+500 -morphology dilate disk +region x1.tiff

magick in.tiff ( +clone -crop 100x100+500+500 -morphology dilate disk ) -layers flatten x2.tiff
```
I think the "region" version applies morphology to the entire image, then internally does a crop and flatten. The second version applies morphology to just the cropped region. It is much faster.

**Reply by Fr4nZ82** [reply]
Thanks! with clone and flatten the cropped tile remains in the right position.
final command:
`convert -density 601.6x600 -units pixelsperinch -monochrome help\1.pdf ( +clone -crop 2500x400+2350+1990 -negate -morphology Thinning "17x17+8+8: -,-,-,-,-,0,0,0,0,0,0,0,-,-,-,-,- -,-,-,-,0,0,0,0,0,0,0,0,0,-,-,-,- -,-,-,0,0,0,0,0,0,0,0,0,0,0,-,-,- -,-,0,0,0,0,-,-,-,-,-,0,0,0,0,-,- -,0,0,0,0,-,-,-,-,-,-,-,0,0,0,0,- 0,0,0,0,-,-,-,-,-,-,-,-,-,0,0,0,0 0,0,0,-,-,-,-,-,-,-,-,-,-,-,0,0,0 0,0,0,-,-,-,-,-,-,-,-,-,-,-,0,0,0 0,0,0,-,-,-,-,-,1,-,-,-,-,-,0,0,0 0,0,0,-,-,-,-,-,-,-,-,-,-,-,0,0,0 0,0,0,-,-,-,-,-,-,-,-,-,-,-,0,0,0 0,0,0,0,-,-,-,-,-,-,-,-,-,0,0,0,0 -,0,0,0,0,-,-,-,-,-,-,-,0,0,0,0,- -,-,0,0,0,0,-,-,-,-,-,0,0,0,0,-,- -,-,-,0,0,0,0,0,0,0,0,0,0,0,-,-,- -,-,-,-,0,0,0,0,0,0,0,0,0,-,-,-,- -,-,-,-,-,0,0,0,0,0,0,0,-,-,-,-,-" -morphology Thinning "13x13+6+6: -,-,0,0,0,0,0,0,0,0,0,-,- -,0,0,0,0,0,0,0,0,0,0,0,- 0,0,0,-,-,-,-,-,-,-,0,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,1,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,-,-,-,-,-,-,-,-,-,0,0 0,0,0,-,-,-,-,-,-,-,0,0,0 -,0,0,0,0,0,0,0,0,0,0,0,- -,-,0,0,0,0,0,0,0,0,0,-,-" -morphology Dilate Disk -morphology Thinning "9x9+4+4: -,-,0,0,0,0,0,-,- -,0,0,0,0,0,0,0,- 0,0,0,-,-,-,0,0,0 0,0,-,-,-,-,-,0,0 0,0,-,-,1,-,-,0,0 0,0,-,-,-,-,-,0,0 0,0,0,-,-,-,0,0,0 -,0,0,0,0,0,0,0,- -,-,0,0,0,0,0,-,-" -morphology Erode Disk -morphology Thinning "9x9+4+4: -,-,0,0,0,0,0,-,- -,0,0,0,0,0,0,0,- 0,0,0,-,-,-,0,0,0 0,0,-,-,-,-,-,0,0 0,0,-,-,1,-,-,0,0 0,0,-,-,-,-,-,0,0 0,0,0,-,-,-,0,0,0 -,0,0,0,0,0,0,0,- -,-,0,0,0,0,0,-,-" -negate ) -layers flatten -compress zip result.pdf`
